test/unit-tests/GAC.t.sol

```diff
@@ -0,0 +1,222 @@
+        gac.setMultiMessageSender(address(42));
+    }
+
+    /// @dev cannot set multi message sender to zero address
+    function test_set_multi_message_sender_zero_address() public {
+        vm.startPrank(owner);
+
+        vm.expectRevert(Error.ZERO_ADDRESS_INPUT.selector);
+        gac.setMultiMessageSender(address(0));
+    }
+
+    /// @dev sets multi message caller
+    function test_set_multi_message_caller() public {
+        vm.startPrank(owner);
+
+        vm.expectEmit(true, true, true, true, address(gac));
+        emit MultiMessageCallerUpdated(address(42));
+
+        gac.setMultiMessageCaller(address(42));
+
+        assertEq(gac.getMultiMessageCaller(), address(42));
+    }
+
+    /// @dev only owner can set multi message caller
+    function test_set_multi_message_caller_only_owner() public {
+        vm.startPrank(caller);
+
+        vm.expectRevert("Ownable: caller is not the owner");
+        gac.setMultiMessageCaller(address(42));
+    }
+
+    /// @dev cannot set multi message caller to zero address
+    function test_set_multi_message_caller_zero_address() public {
+        vm.startPrank(owner);
+
+        vm.expectRevert(Error.ZERO_ADDRESS_INPUT.selector);
+        gac.setMultiMessageCaller(address(0));
+    }
+
+    /// @dev sets multi message receiver
+    function test_set_multi_message_receiver() public {
+        vm.startPrank(owner);
+
+        vm.expectEmit(true, true, true, true, address(gac));
+        emit MultiMessageReceiverUpdated(DST_CHAIN_ID, address(42));
+
+        gac.setMultiMessageReceiver(DST_CHAIN_ID, address(42));
+
+        assertEq(gac.getMultiMessageReceiver(DST_CHAIN_ID), address(42));
+    }
+
+    /// @dev only owner can set multi message receiver
+    function test_set_multi_message_receiver_only_owner() public {
+        vm.startPrank(caller);
+
+        vm.expectRevert("Ownable: caller is not the owner");
+        gac.setMultiMessageReceiver(DST_CHAIN_ID, address(42));
+    }
+
+    /// @dev cannot set multi message receiver to zero address
+    function test_set_multi_message_receiver_zero_address() public {
+        vm.startPrank(owner);
+
+        vm.expectRevert(Error.ZERO_ADDRESS_INPUT.selector);
+        gac.setMultiMessageReceiver(DST_CHAIN_ID, address(0));
+    }
+
+    /// @dev cannot set multi message receiver on zero chain ID
+    function test_set_multi_message_receiver_zero_chain_id() public {
+        vm.startPrank(owner);
+
+        vm.expectRevert(Error.ZERO_CHAIN_ID.selector);
+        gac.setMultiMessageReceiver(0, address(42));
+    }
+
+    /// @dev sets global message delivery gas limit
+    function test_set_global_msg_delivery_gas_limit() public {
+        vm.startPrank(owner);
+
+        uint256 oldLimit = gac.getGlobalMsgDeliveryGasLimit();
+        vm.expectEmit(true, true, true, true, address(gac));
+        emit DstGasLimitUpdated(oldLimit, 420000);
+
+        gac.setGlobalMsgDeliveryGasLimit(420000);
```

Comment:
I suggest we have a sensible lower-bound for what this value can be, and validate changes accordingly.

Reply:
Done.